src/Form/ContentpoolChannelForm.php

```diff
@@ -0,0 +1,148 @@
+<?php
+
+namespace Drupal\contentpool_client\Form;
+
+use Drupal\contentpool_client\RemotePullManagerInterface;
+use Drupal\Core\Access\AccessResult;
+use Drupal\Core\Form\FormBase;
+use Drupal\Core\Form\FormStateInterface;
+use Drupal\Core\Messenger\MessengerInterface;
+use Drupal\Core\Session\AccountInterface;
+use Drupal\relaxed\Entity\RemoteInterface;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+
+/**
+ * Class ContentpoolChannelForm.
+ */
+class ContentpoolChannelForm extends FormBase {
+
+  /**
+   * The Messenger service.
+   *
+   * @var \Drupal\Core\Messenger\MessengerInterface
+   */
+  protected $messenger;
+
+  /**
+   * The remote pull manager.
+   *
+   * @var \Drupal\contentpool_client\RemotePullManagerInterface
+   */
+  protected $remotePullManager;
+
+  /**
+   * Constructs the object.
+   */
+  public function __construct(MessengerInterface $messenger, RemotePullManagerInterface $remote_pull_manager) {
+    $this->messenger = $messenger;
+    $this->remotePullManager = $remote_pull_manager;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function create(ContainerInterface $container) {
+    return new static(
+      $container->get('messenger'),
+      $container->get('contentpool_client.remote_pull_manager')
+    );
+  }
+
+  /**
+   * Returns a unique string identifying the form.
+   *
+   * @return string
+   *   The unique string identifying the form.
+   */
+  public function getFormId() {
+    return 'remote_contentpool_channels';
+  }
+
+  /**
+   * Form constructor.
+   *
+   * @param array $form
+   *   An associative array containing the structure of the form.
+   * @param \Drupal\Core\Form\FormStateInterface $form_state
+   *   The current state of the form.
+   * @param \Drupal\relaxed\Entity\RemoteInterface|null $remote
+   *   The remote entity.
+   *
+   * @return array
+   *   The form structure.
+   */
+  public function buildForm(array $form, FormStateInterface $form_state, RemoteInterface $remote = NULL) {
+    // Verify the connection to the remote.
+    // Add channel subscription settings.
+    list($channels, $topics) = $this->remotePullManager->getChannelOptions($remote);
+    $channel_uuids = $remote->getThirdPartySetting('contentpool_client', 'channels', []);
+
+    if (empty($channels)) {
+      $this->messenger->addWarning('No channels from contentpool server available.');
+    }
+
+    $form_state->set('remote', $remote);
+
+    $form['contentpool_channels'] = [
+      '#type' => 'checkboxes',
+      '#title' => $this->t('Content channels'),
+      '#description' => $this->t('Channels of content that this site is subscribed to.'),
+      '#options' => $channels,
+      '#default_value' => $channel_uuids,
+      '#required' => TRUE,
+    ];
+
+    // Add topic subscription settings.
+    $topic_uuids = $remote->getThirdPartySetting('contentpool_client', 'topics', []);
+
+    $form['contentpool_topics'] = [
```

Comment:
We may not have topics here, it should be generic to whatever term reference field is on the article.